api/include/opentelemetry/trace/propagation/httptextformat.h

```diff
@@ -10,21 +11,21 @@ namespace trace
 namespace propagation
 {
 
-virtual static Context SetSpanInContext(Span span, Context context = NULL)
-virtual static Span GetCurrentSpan(Context context = NULL)
+virtual static Context SetSpanInContext(Span span, Context &context = NULL)
+virtual static Span GetCurrentSpan(Context &context = NULL)
 
+// The HTTPTextFormat class provides an interface that enables extracting and injecting
+// context into headers of HTTP requests. HTTP frameworks and clients
+// can integrate with HTTPTextFormat by providing the object containing the
+//  headers, and a getter and setter function for the extraction and
+// injection of values, respectively.
 template <typename T>
 class HTTPTextFormat {
-    /** This class provides an interface that enables extracting and injecting
-    context into headers of HTTP requests. HTTP frameworks and clients
-    can integrate with HTTPTextFormat by providing the object containing the
-     headers, and a getter and setter function for the extraction and
-    injection of values, respectively.*/
     public:
-        using Setter = nostd::string_view(*)(nostd::string_view,nostd::string_view);
-        using Getter = void(*)(T&,nostd::string_view,nostd::string);
-        virtual Context extract(Setter get_from_carrier, const T &carrier, Context &context)
-        virtual void inject(Getter set_from_carrier, T &carrier, const Context &context)
+        using Getter = nostd::string_view(*)(T &carrier, nostd::string_view trace_type);
+        using Setter = void(*)(T &carrier, nostd::string_view trace_type,nostd::string trace_description);
+        virtual Context extract(Getter get_from_carrier, const T &carrier, Context &context)
+        virtual void inject(Setter set_from_carrier, T &carrier, const Context &context)
```

Comment:
I think you want to make the methods pure virtual with the expectation that subclasses define them.

```suggestion
        virtual Context extract(Getter get_from_carrier, const T &carrier, Context &context) = 0;
        virtual void inject(Setter set_from_carrier, T &carrier, const Context &context) = 0;
```